server/WebAPI/Services/ScheduledPlaceToEatService.cs

```diff
@@ -0,0 +1,254 @@
+    private readonly IScheduledPlaceToEatRepository _scheduledPlaceToEatRepository;
+    private readonly IAttachmentFileEatingRepository _attachmentFileEatingRepository;
+
+    public ScheduledPlaceToEatService(
+      UserService userService,
+      IScheduledPlaceToEatRepository scheduledPlaceToEatRepository,
+      IAttachmentFileEatingRepository attachmentFileEatingRepository)
+    {
+      _userService = userService;
+      _scheduledPlaceToEatRepository = scheduledPlaceToEatRepository;
+      _attachmentFileEatingRepository = attachmentFileEatingRepository;
+    }
+
+    public int CreateNewEating(InputScheduledPlaceToEatForCreateDTO scheduledPlaceToEatDTO)
+    {
+      try
+      {
+        if(!Valid(scheduledPlaceToEatDTO))
+        {
+          throw new ArgumentException();
+        }
+        else
+        {
+          if(_userService.GetUser(scheduledPlaceToEatDTO.UserId) != null)
+          {
+            return _scheduledPlaceToEatRepository.CreateScheduledPlaceToEat(ConvertInputScheduledPlaceToEatForCreateDTOToScheduletPlaceToEat(scheduledPlaceToEatDTO));
+          }
+          else
+          {
+            throw new ArgumentException();
+          }
+        }
+      }
+      catch(Exception)
+      {
+        return -1;
+      }
+    }
+
+    public IEnumerable<OutputScheduledPlaceToEatDTO> GetEatingByUserId(int UserId)
+    {
+      if(_userService.GetUser(UserId) != null)
+      {
+        var eatings = _scheduledPlaceToEatRepository.GetScheduledPlaceToEatByUserId(UserId);
+        List<OutputScheduledPlaceToEatDTO> result = new List<OutputScheduledPlaceToEatDTO>();
+        foreach(var eating in eatings)
+        {
+          result.Add(ConvertScheduledPlaceToEatToOutputScheduletPlaceToEatDTO(eating));
+        }
+        return result;
+      }
+      else
+      {
+        return null;
+      }
+    }
+
+    public bool UpdateScheduledPlaceToEat(InputScheduledPlaceToEatForUpdateOrDeleteDTO scheduledPlaceToEatDTO)
+    {
+      try
+      {
+        if(!Valid(scheduledPlaceToEatDTO))
+        {
+          throw new ArgumentException();
+        }
+        else
+        {
+          var oldEating = _scheduledPlaceToEatRepository.GetScheduledPlaceToEatById(scheduledPlaceToEatDTO.Id);
+          if( oldEating == null)
+          {
+            throw new ArgumentException();
+          }
+          oldEating.Notes = scheduledPlaceToEatDTO.Notes;
+          oldEating.NamePlace = scheduledPlaceToEatDTO.NamePlace;
+          oldEating.Lng = scheduledPlaceToEatDTO.Lng;
+          oldEating.Lat = scheduledPlaceToEatDTO.Lat;
+          oldEating.Link = scheduledPlaceToEatDTO.Link;
+          oldEating.GooglePlaceId = scheduledPlaceToEatDTO.GooglePlaceId;
+          oldEating.DateTime = scheduledPlaceToEatDTO.DateTime;
+          _scheduledPlaceToEatRepository.UptateScheduledPlaceToEat(oldEating);
+          return true;
+        }
+      }
+      catch(Exception e)
+      {
+        Console.WriteLine(e);
```

Comment:
any message for user why this didn't worked out?

Reply:
ok